clients/src/test/java/org/apache/kafka/clients/consumer/internals/AsyncKafkaConsumerTest.java

```diff
@@ -463,6 +467,42 @@ public void onPartitionsAssigned(final Collection<TopicPartition> partitions) {
         assertTrue(callbackExecuted.get());
     }
 
+    @Test
+    public void testCheckForNewTopicOnlyWhenMetadataChange() {
+        SubscriptionState subscriptions = mock(SubscriptionState.class);
+        Cluster cluster = mock(Cluster.class);
+
+        consumer = newConsumer(
+                mock(FetchBuffer.class),
+                mock(ConsumerInterceptors.class),
+                mock(ConsumerRebalanceListenerInvoker.class),
+                subscriptions,
+                singletonList(new RoundRobinAssignor()),
```

Comment:
we should not have to deal with the old assignors in the new consumer, so this made me notice we have a bug that makes you have to provide this. That needs to be fixed, so I filed https://issues.apache.org/jira/browse/KAFKA-16786 for it.

Reply:
This PR got merged so let's get the latest changes to get rid of the deprecated assignor here

Reply:
I understand, will include the change in the next commit